src/scripts/emailUpperCaseMigration.js

```diff
@@ -0,0 +1,145 @@
+import createDbConnection from "../db/connect.js";
+const knex = createDbConnection();
+
+import { subscribeHash } from "../utils/hibp.js";
+import { getSha1 } from "../utils/fxa.js";
+
+/**
+ * MNTOR-2469: One-off script to lowercase email addresses of subscribers and emails before hashing
+ */
+async function subscribeLowerCaseHashToHIBP(emailAddress) {
+  const lowerCasedEmail = emailAddress.toLowerCase();
+  const lowerCasedSha1 = getSha1(lowerCasedEmail);
+  await subscribeHash(lowerCasedSha1);
+  return lowerCasedSha1;
+}
+
+(async () => {
+  const chunkSize = 1000;
+  console.log(
+    `Fixing lower-cased emails / sub to hashes in ${chunkSize}-sized chunks`,
+  );
+
+  console.log(`/** Subscribers Table **/`);
+  const subRecordsThatNeedFixing = await knex("subscribers")
+    .count()
+    .whereRaw("primary_email != lower(primary_email)");
+  const subsWithUpperCount = subRecordsThatNeedFixing[0].count;
+  console.log(
+    `found ${subsWithUpperCount} subscribers records with primary_email != lower(primary_email)`,
+  );
+
+  let subRecordsFixed = 0;
+  let subPrevMaxId = 0;
+  while (subRecordsFixed < subsWithUpperCount) {
+    console.log(
+      `working on ${chunkSize}-sized chunk where id > ${subPrevMaxId} ...`,
+    );
+    const subRecordsWithUpperCharsChunk = await knex
+      .select("id", "primary_email")
+      .from("subscribers")
+      .where("id", ">", subPrevMaxId)
+      .whereRaw("primary_email != lower(primary_email)")
```

Comment:
Too bad knex doesn't have support for this but I think this is the best approach.